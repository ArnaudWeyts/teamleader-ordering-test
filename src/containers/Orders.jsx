import React, { Component } from 'react';
import { connect } from 'react-redux';

import { ordersType } from '../types';

import { fetchOrders } from '../actions/ordersActions';

import Order from '../components/Order';

class Orders extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchOrders());
  }

  render() {
    const {
      orders: { allOrders, isFetching }
    } = this.props;

    if (isFetching) {
      return <div>Loading...</div>;
    }

    return (
      <div>
        <h2>Orders</h2>
        {allOrders.map(order => (
          <Order key={order.id} order={order} />
        ))}
      </div>
    );
  }
}

Orders.propTypes = ordersType;

const mapStateToProps = state => ({
  orders: state.orders
});

export default connect(mapStateToProps)(Orders);
