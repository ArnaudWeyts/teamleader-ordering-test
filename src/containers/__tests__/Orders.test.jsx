import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';

import Orders from '../Orders';

const mockStore = configureStore([thunk]);

it('renders without crashing', () => {
  const store = mockStore({ orders: { allOrders: [] } });
  shallow(<Orders store={store} />);
});

it('renders the orders loading', () => {
  const store = mockStore({
    orders: {
      allOrders: []
    }
  });

  const component = renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <Orders store={store} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the orders', () => {
  const store = mockStore({
    orders: {
      allOrders: [
        {
          id: '1',
          'customer-id': '1',
          items: [
            {
              'product-id': 'B102',
              quantity: 10,
              'unit-price': 4.99,
              total: 49.9
            }
          ],
          total: 49.9
        },
        {
          id: '2',
          'customer-id': '2',
          items: [
            {
              'product-id': 'B102',
              quantity: 5,
              'unit-price': 4.99,
              total: 24.95
            }
          ],
          total: 24.95
        }
      ]
    }
  });

  const component = renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <Orders store={store} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
