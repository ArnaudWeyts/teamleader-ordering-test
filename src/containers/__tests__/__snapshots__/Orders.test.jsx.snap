// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the orders 1`] = `
<div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <h3>
        Order #
        1
      </h3>
      <a
        href="/orders/1"
        onClick={[Function]}
      >
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="button"
        >
          <span>
            View
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <h3>
        Order #
        2
      </h3>
      <a
        href="/orders/2"
        onClick={[Function]}
      >
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="button"
        >
          <span>
            View
          </span>
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders the orders loading 1`] = `<div />`;
